### ISSUE TITLE
Generate data from a specified AUC

### ISSUE THREAD

**Issue by kgoldfeld** [kgoldfeld/simstudy#126]
A user wrote in with this question: **I wonder whether you have an option in this package to simulate data from a given AUC (area under the roc curve)?**

There is currently no such option, but a solution has been described [here](https://stats.stackexchange.com/questions/422926/generate-synthetic-data-given-auc), and that solution includes a reference to this [paper](https://journals.copmadrid.org/ejpalc/art/ejpalc2018a5).

And here is the code from the solution:

```r
auc <- 0.95

t <- sqrt(log(1/(1-auc)**2))
z <- t-((2.515517 + 0.802853*t + 0.0103328*t**2) / 
          (1 + 1.432788*t + 0.189269*t**2 + 0.001308*t**3))
d <- z*sqrt(2)

n <- 10000
x <- c(rnorm(n/2, mean = 0), rnorm(n/2, mean = d))
y <- c(rep(0, n/2), rep(1, n/2))

library(AUC)
auc(roc(x, as.factor(y)))

# out
# [1] 0.9486257
```

**Comment by assignUser** [comment]
I will have to look into this a bit more but I would like to not add the dependency if possible. Maybe we can re-implement `AUC::auc()`

**Comment by kgoldfeld** [comment]
I definitely wasn't thinking of implementing exactly like this - and the use of `AUC::auc()` was merely to check the results - it is not part of the algorithm. There are more thorough algorithms that allow covariates, which is why the initial question arose. I have a number of papers that describe these, but haven't read through them yet. I'd put this fairly low on the priority scale.

**Comment by assignUser** [comment]
Ok, I'll focus on the other issues for now :)

**Comment by kgoldfeld** [comment]
This has been implemented in function `logisticCoefs` and is available as of the development version `0.7.0.9000`.
